File: content/problems/1400-construct-k-palindrome-strings/solutions/odd-frequency-bound/Solution.kt
class Solution {
    fun canConstruct(text:String,k:Int):Boolean{
        if(text.length<k)return false;
        val count=IntArray(26);
        for(value in text)count[value-'a']++;
        return count.count{
            it%2==1
        }<=k
    }
}

File: content/problems/1400-construct-k-palindrome-strings/solutions/odd-frequency-bound/solution.cpp
class Solution {
    public: bool canConstruct(string text,int k){
        if((int)text.size()<k)return false;
        int count[26]={
        };
        for(char value:text)++count[value-'a'];
        int odd=0;
        for(int frequency:count)odd+=frequency%2;
        return odd<=k;
    }
};
